File: code/backend/rust/common/src/search_doc.rs
use std::fmt::Debug;

use serde::Deserialize;
use serde::Serialize;

#[derive(Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SearchIndexDoc {
    pub id: String,
    pub tconst: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub titles: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub actors: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub directors: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub writers: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub runtime_minutes: Option<u32>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub adult: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub genres: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub characters: Option<Vec<String>>,
}

File: code/backend/csharp/SearchIndexService/SearchIndexService/SearchDoc.cs
namespace SearchIndexService;


public class SearchDoc
{
    public string tconst { get; set; }

    public List<string>? Titles { get; set; }
    public List<string>? Actors { get; set; }
    public List<string>? Directors { get; set; }
    public List<string>? Genres { get; set; }
    public List<string>? Characters { get; set; }

    public int? RuntimeMinutes { get; set; }
    public int? Year { get; set; }

    public bool? Adult { get; set; }
    public string? TitleType { get; set; }
}